# Setup

In [6]:
!pip install requests

  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
  Using cached certifi-2023.7.22-py3-none-any.whl (158 kB)
  Using cached idna-3.4-py3-none-any.whl (61 kB)
  Using cached urllib3-2.0.4-py3-none-any.whl (123 kB)
  Using cached charset_normalizer-3.2.0-cp310-cp310-macosx_10_9_x86_64.whl (126 kB)

[notice] A new release of pip is available: 23.0.1 -> 23.2.1
[notice] To update, run: pip install --upgrade pip


# CURL

* **List users**

In [2]:

!curl -i -H "Accept:application/json" -H "Content-Type:application/json" -H "Authorization: Bearer YOUR-TOKEN-HERE" -XGET "https://gorest.co.in/public/v2/users"


HTTP/2 200 
date: Tue, 08 Aug 2023 15:36:30 GMT
content-type: application/json; charset=utf-8
cache-control: max-age=0, private, must-revalidate
etag: W/"d5801368b8652b64056a30e734e4a0c6"
referrer-policy: strict-origin-when-cross-origin
vary: Origin
x-content-type-options: nosniff
x-download-options: noopen
x-frame-options: SAMEORIGIN
x-links-current: https://gorest.co.in/public/v2/users?page=1
x-links-next: https://gorest.co.in/public/v2/users?page=2
x-links-previous: 
x-pagination-limit: 10
x-pagination-page: 1
x-pagination-pages: 275
x-pagination-total: 2748
x-permitted-cross-domain-policies: none
x-ratelimit-limit: 90
x-ratelimit-remaining: 89
x-ratelimit-reset: 1
x-request-id: 72bd319c-0ec2-44d1-b6e1-00c80fac9cb8
x-runtime: 0.047090
x-xss-protection: 0
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=V4WwW9daUhMDC7w9frn%2FMioRRCvv3H6CPoPyvhUWGfDIWyQp4Uf%2FCmJdrjJsjn%2FikS5c0gZvez57CO6Cs5Dz47raz0a3W6S8%2BLM14MFu%2BUsOjBDxD2Sf0hA

* **Create user**

In [4]:
!curl -i -H "Accept:application/json" -H "Content-Type:application/json" -H "Authorization: Bearer YOUR-TOKEN-HERE" -XPOST "https://gorest.co.in/public/v2/users" -d '{"name":"Tenali Ramakrishna", "gender":"male", "email":"carlos@15ce.com", "status":"active"}'

HTTP/2 201 
date: Tue, 08 Aug 2023 15:37:19 GMT
content-type: application/json; charset=utf-8
location:]8;;https://gorest.co.in/public/v2/users/4297712\ https://gorest.co.in/public/v2/users/4297712
]8;;\cache-control: max-age=0, private, must-revalidate
etag: W/"c27cd90cfe619b992b3d38bc0ffda284"
referrer-policy: strict-origin-when-cross-origin
vary: Origin
x-content-type-options: nosniff
x-download-options: noopen
x-frame-options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-ratelimit-limit: 90
x-ratelimit-remaining: 89
x-ratelimit-reset: 1
x-request-id: b2cffb2d-decf-48cd-bd8f-5bf19f28b07e
x-runtime: 0.072919
x-xss-protection: 0
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=UKqeI9vygt%2BqMdy6Um%2Bv8Bxs%2BmZaM8vKspHeSq8G5xhOoktQw4pQpq%2B%2BqtF7IbnQostJ0vLfj0hdMf25stLPuUCO6hbQYwmbUx8ClZah%2BHxqNR0mj4u0rs%2Fx0qa9HL1Kj28tT0rMPqz7yUg%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel",

# Python Requests

* **Update user**

In [9]:
import requests

url = "https://gorest.co.in/public/v2/users/2481"
headers = {
    "Accept": "application/json",
    "Content-Type": "application/json",
    "Authorization": "Bearer YOUR-TOKEN-HERE"
}
data = {
    "name": "Allasani Peddana",
    "email": "carlos_l@15ce.com",
    "status": "active"
}

response = requests.patch(url, headers=headers, json=data)

print(response.status_code)
print(response.text)

200
{"email":"carlos_l@15ce.com","name":"Allasani Peddana","status":"active","id":2481,"gender":"female"}


* **Delete user**

In [10]:
import requests

url = "https://gorest.co.in/public/v2/users/2481"
headers = {
    "Accept": "application/json",
    "Content-Type": "application/json",
    "Authorization": "Bearer YOUR-TOKEN-HERE"
}

response = requests.delete(url, headers=headers)

print(response.status_code)
print(response.text)

204



# REST API Http Response Codes
* 200: OK. Everything worked as expected.
* 201: A resource was successfully created in response to a POST request. The Location header contains the URL pointing to the newly created resource.
* 204: The request was handled successfully and the response contains no body content (like a DELETE request).
* 304: The resource was not modified. You can use the cached version.
* 400: Bad request. This could be caused by various actions by the user, such as providing invalid JSON data in the request body etc.
* 401: Authentication failed.
* 403: The authenticated user is not allowed to access the specified API endpoint.
* 404: The requested resource does not exist.
* 405: Method not allowed. Please check the Allow header for the allowed HTTP methods.
* 415: Unsupported media type. The requested content type or version number is invalid.
* 422: Data validation failed (in response to a POST request, for example). Please check the response body for detailed error messages.
* 429: Too many requests. The request was rejected due to rate limiting.
* 500: Internal server error. This could be caused by internal program errors.